chessington/engine/pieces.py

```diff
@@ -33,14 +34,90 @@ class Pawn(Piece):
     """
     A class representing a chess pawn.
     """
+
     def get_available_moves(self, board) -> List[Square]:
         current_square = board.find_piece(self)
+        square_list = []
+        if not self.can_move(board, current_square):
+            return square_list
+
+        if self.player == Player.BLACK:
+            square_list.append(Square.at(current_square.row - 1, current_square.col))
+
+            if current_square.row == 6 and not self.has_piece_in_front(board, current_square, distance=2):
+                square_list.append(Square.at(current_square.row - 2, current_square.col))
+        else:
+            square_list.append(Square.at(current_square.row + 1, current_square.col))
+            if current_square.row == 1 and not self.has_piece_in_front(board, current_square, distance=2):
+                square_list.append(Square.at(current_square.row + 2, current_square.col))
+        square_list += self.get_moves_diag(board, current_square)
+
+        return square_list
+
+    def has_piece_in_front(self, board: Board, current_square, distance=1) -> bool:
+        if self.player == Player.BLACK:
+            if board.get_piece(Square.at(current_square.row - distance, current_square.col)):
+                return True
+        else:
+            if board.get_piece(Square.at(current_square.row + distance, current_square.col)):
+                return True
+
+        return False
+
+    def at_edge(self, square: Square) -> bool:
+        if self.player == Player.BLACK:
+            if square.row == 0:
+                return True
+        else:
+            if square.row == 7:
+                return True
+
+        return False
+
+    def has_piece_diagonal(self, board: Board, square: Square) -> bool:
+        if self.player == Player.BLACK:
+            diag_left = board.get_piece(Square.at(square.row - 1, square.col - 1))
+            diag_right = board.get_piece(Square.at(square.row - 1, square.col + 1))
+
+            if (diag_left and diag_left.player == Player.WHITE) \
+                    or (diag_right and diag_right.player == Player.BLACK):
+                return True
+        else:
+            diag_left = board.get_piece(Square.at(square.row + 1, square.col - 1))
+            diag_right = board.get_piece(Square.at(square.row + 1, square.col + 1))
+
+            if (diag_left and diag_left.player == Player.BLACK) \
+                    or (diag_right and diag_right.player == Player.BLACK):
+                return True
+        return False
+
+    def get_moves_diag(self, board: Board, square: Square) -> List[Square]:
```

Comment:
I don't understand why have both `has_piece_diagonal` and `get_moves_diagonal` ?

Reply:
da pentru ca sunt fraier si fac intai verificari daca pot muta si dupa vad ce pot sa mut :(